From e3db9462f872b80941fb6f6bbec78f9fa1881297 Mon Sep 17 00:00:00 2001
From: Finley Xiao <finley.xiao@rock-chips.com>
Date: Mon, 8 Jan 2018 19:50:13 +0800
Subject: [PATCH] arm64: dts: rockchip: rk3328: Add rkvdec and vpu qos node

The rkvdec and vpu qos registers need to save and restore when reset.

Change-Id: If0fbee0aed9227cfd795c5f439cfb8c3b2f0ccaf
Signed-off-by: Finley Xiao <finley.xiao@rock-chips.com>
---
 arch/arm64/boot/dts/rockchip/rk3328.dtsi | 17 +++++++++++++++++
 1 file changed, 17 insertions(+)

diff --git a/arch/arm64/boot/dts/rockchip/rk3328.dtsi b/arch/arm64/boot/dts/rockchip/rk3328.dtsi
index dd15ca020abb..448421a43aca 100644
--- a/arch/arm64/boot/dts/rockchip/rk3328.dtsi
+++ b/arch/arm64/boot/dts/rockchip/rk3328.dtsi
@@ -305,9 +305,11 @@
 			};
 			pd_video@RK3328_PD_VIDEO {
 				reg = <RK3328_PD_VIDEO>;
+				pm_qos = <&qos_rkvdec_r>, <&qos_rkvdec_w>;
 			};
 			pd_vpu@RK3328_PD_VPU {
 				reg = <RK3328_PD_VPU>;
+				pm_qos = <&qos_vpu>;
 			};
 		};
 
@@ -1347,6 +1349,21 @@
 		};
 	};
 
+	qos_rkvdec_r: qos@ff750000 {
+		compatible = "syscon";
+		reg = <0x0 0xff750000 0x0 0x20>;
+	};
+
+	qos_rkvdec_w: qos@ff750080 {
+		compatible = "syscon";
+		reg = <0x0 0xff750080 0x0 0x20>;
+	};
+
+	qos_vpu: qos@ff778000 {
+		compatible = "syscon";
+		reg = <0x0 0xff778000 0x0 0x20>;
+	};
+
 	dfi: dfi@ff790000 {
 		reg = <0x00 0xff790000 0x00 0x400>;
 		compatible = "rockchip,rk3328-dfi";
-- 
2.35.3

